From 630328ef07379bc294c9f284cf878f68540b2daa Mon Sep 17 00:00:00 2001
From: Casey Schaufler <casey@schaufler-ca.com>
Date: Tue, 1 Sep 2020 11:19:11 -0700
Subject: [PATCH 31/36] UBUNTU: SAUCE: Audit: Fix for missing NULL check

AppArmor audit calls can have a NULL audit context,
so the LSM context audit needs to check for this.

Signed-off-by: Casey Schaufler <casey@schaufler-ca.com>
Signed-off-by: John Johansen <john.johansen@canonical.com>
Acked-by: Andrea Righi <andrea.righi@canonical.com>
Signed-off-by: Seth Forshee <seth.forshee@canonical.com>
---
 kernel/audit.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/kernel/audit.c b/kernel/audit.c
index 8b76beea04dad..4770f836e97c7 100644
--- a/kernel/audit.c
+++ b/kernel/audit.c
@@ -1930,7 +1930,8 @@ struct audit_buffer *audit_log_start(struct audit_context *ctx, gfp_t gfp_mask,
 	/* cancel dummy context to enable supporting records */
 	if (ctx)
 		ctx->dummy = 0;
-	if (type == AUDIT_MAC_TASK_CONTEXTS && ab->ctx->serial == 0) {
+	if (type == AUDIT_MAC_TASK_CONTEXTS && ab->ctx &&
+	    ab->ctx->serial == 0) {
 		audit_stamp_context(ab->ctx);
 		audit_get_stamp(ab->ctx, &t, &serial);
 	}
-- 
2.38.1

